etna/auto/optuna/config_sampler.py

```diff
@@ -0,0 +1,140 @@
+from typing import List
+from typing import Optional
+from typing import Set
+
+import numpy as np
+from optuna.samplers import BaseSampler
+from optuna.study import Study
+from optuna.trial import FrozenTrial
+from optuna.trial import TrialState
+
+from etna.auto.utils import config_hash
+from etna.auto.utils import retry
+
+
+class ConfigSampler(BaseSampler):
+    """Optuna based sampler for greedy search over different configurations."""
+
+    def __init__(self, configs: List[dict], random_generator: Optional[np.random.Generator] = None, retries: int = 10):
+        """Init Config sampler.
+
+        Parameters
+        ----------
+        configs:
+            pool of configs to sample from
+        random_generator:
```

Comment:
Do we really need `random_generator` and `retries` parameters to be able to set? They seem to be some technical setting that does not affect the general behavior of the sampler -- it still sample all the configs from the `configs` list. May be it's child i.e. RandomSampler needs random_generator to be explicitly set

Reply:
I've added it for easier testing: to fix generator and get reproducible for example in the future.

Reply:
This parameters might be confusing if you don't now the implementation details, let's create a setter method for them, if someone really want to set them customary, he would.